File: 00_pilot/Gyeongwook/combination.py
class Solution:
    def build(self, item, currentLen, start, end, targetLen, returnValue):
        if start > end :
            return

        for eachNum in range(start, end) :
            newItem = []
            for temp in item :
                newItem.append(temp)

            newItem.append(eachNum)
            len = currentLen + 1

            if targetLen - len <= 0 :
                returnValue.append(newItem)
            else :
                self.build(newItem, len, eachNum + 1, end, targetLen, returnValue)
                
    # @return a list of lists of integers
    def combine(self, n, k):
        returnVal = []
        currentItem = []
        self.build(currentItem, 0, 1, n + 1, k, returnVal)

        return returnVal

File: 00_pilot/Eunbeom/combination.cpp
class Solution {
public:
    
    vector<vector<int>> vv;
    vector<int> v;
    
    vector<vector<int> > combine(int n, int k) {
        f(n, k, 0);
        return vv;
    }
    
    void f(int n, int k, int i){
        
        if(k==0){
            vv.push_back(v);
            return;
        }

        for(int j=i+1;j<=n;j++){
            v.push_back(j);
            f(n, k-1, j);
            v.pop_back();
        }
    }
};